data/initial-composition/slab-model/create_slab2_database.py

```diff
@@ -0,0 +1,178 @@
+# M., & Smoczyk, G. M.  (2018). Slab2,a comprehensive subduction zone geometry
+# model. Science, 362(6410), 58-61.
+# 
+# Output: A structured file containing all the slabs for the Earth with columns
+# representing longitude, colatitude, slab depths, and slab thickness
+
+import numpy as np
+import os
+from glob import glob
+from scipy.interpolate import griddata
+
+# Save the current working directory
+cwd = os.getcwd()
+
+# Set the path to the downloaded slab2 model here. This path is correct if you
+# download the model from the address above and extract in the current folder.
+os.chdir('./Slab2Distribute_Mar2018/Slab2_TXT')
+
+
+def interpolate_data (name_key, type_key):
+    """
+    From initial data inspection, we found that most slabs have spacing of 0.05 degrees except for 5 slabs,   
+    which have 0.02 degree spacing. Therefore, this function interpolates those slabs to 0.05 spacing for 
+    consistency.
+    """
+
+    file_name = glob(name_key + '*' + type_key +'*.xyz')
+    data_in   = np.loadtxt (file_name[0], delimiter=",")
+    
+    # We convert latitudes to co-latitudes for all the slab files.
+    points_x  = data_in[:, 0] 
+    points_y  = 90 - data_in[:, 1]
+    field     = data_in[:, 2]
+    
+    x0 = np.min(points_x)
+    x1 = np.max(points_x)
+    y0 = np.min(points_y)
+    y1 = np.max(points_y)
+    
+    # create the desired mesh
+    x = np.arange (x0, x1 + 0.05, 0.05)
+    y = np.arange (y0, y1 + 0.05, 0.05)
+    
+    x_grid, y_grid = np.meshgrid (x, y)
+    
+    depth_interp = griddata ((points_x, points_y), field, (x_grid, y_grid), method = 'linear')
+    data_save    = np.column_stack ((x_grid.flatten(), 90 - y_grid.flatten(), depth_interp.flatten()))
+    
+    np.savetxt (file_name[0], data_save, fmt='%1.2f', delimiter=',')
+
+
+# Slabs with 0.02 degree spacing
+file_name_key = ['hin', 'man', 'mue', 'pam', 'puy']
+file_type_key = ['dep', 'thk', 'dip']
+
+for i in range(len(file_name_key)):
+    for j in range(len(file_type_key)):
+        interpolate_data (file_name_key[i], file_type_key[j])
+
+# Create the grid for aspect, with global longitudes and colatitudes and
+# resolution same as all the slabs (0.05 degrees)
+longitudes_target = np.arange (0 , 360.05, 0.05)
+latitudes_target  = np.arange (0,  180.05, 0.05)
+longitude_grid, latitude_grid = np.meshgrid (longitudes_target, latitudes_target)
+
+# We initialize the depth array with a very large number and the
+# thickness value with 0 to make it easy to determine regions
+# without slabs.
+depths_target       = np.finfo(np.float64).max / 1e4 * np.ones (len(longitude_grid.flatten()),)
+thickness_target    = np.zeros (len(longitude_grid.flatten()),)
+dips_target         = np.zeros (len(longitude_grid.flatten()),)
+
+nan_indx = 0 
+def slab2_into_asciiboundary (slab_filename):
+    """
+    This function uses the above target mesh and fills it with the input slab depths and thicknesses. 
```

Comment:
You also need to update the documentation here: 
```suggestion
    This function uses the above target mesh and fills it with the input slab depths, thicknesses and dip angles. 
```